src/libtorch.cc

```diff
@@ -512,6 +544,15 @@ ModelInstanceState::ModelInstanceState(
 
 ModelInstanceState::~ModelInstanceState()
 {
+  // If this is the last instance, remove the pointer from the model map
+  if (torch_model_.use_count() == 2) {
+    if (StateForModel()->UnloadModel(
+            std::make_pair(!device_.is_cpu(), device_.index())) != 1) {
+      LOG_MESSAGE(
+          TRITONSERVER_LOG_ERROR,
+          ("Model mapping not found for '" + Name() + "'").c_str());
```

Comment:
"No model found on target device". I would suggest to log device type and device id here for readability.

Technically the mapping is not to a single model instance but rather a model shared across instances. So this statement is incorrect.